### PR TITLE
Custom Panels/Buttons and easier DropdownPanels

### PR DESCRIPTION
**Pull request by AlexGilleran**
This adds a `customElements` prop to `StandardUserInterface` that allows extra elements to be injected into the standard user interface without completely recreating it. This allows for #1659 to be fixed in NationalMap (so it can inject its own related maps panel and about button) and also paves the way for #1694 using the same mechanism.

Is this the right way to open up `StandardUserInterface`? I'm not sure, but it works for now and we'll learn by doing it.

In addition, because the idea is that client maps can inject their own `DropdownPanel`s, I've updated this to use javascript to figure out where to place itself and where to place the caret that points at its button. Without this we have to plug exactly where the panel will be into the CSS, which makes it annoying to add buttons/panels to the right of it.

I've also made the leaflet styles global (which they should've been originally).

For a commit that uses this see https://github.com/NICTA/nationalmap/pull/276.

### CODE REVIEW

File: buildprocess/configureWebpack.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var path = require('path');
 var StringReplacePlugin = require("string-replace-webpack-plugin");
 
 // If node-sass starts hanging, uncomment this line:
-//process.env.UV_THREADPOOL_SIZE = 128;
+process.env.UV_THREADPOOL_SIZE = 128;
```
Comment (kring) [inline buildprocess/configureWebpack.js:5]: Did you mean to commit this?  I don't have a problem with it, just want to make sure.

**Comment by kring** [comment]
This looks great!

There's a small z-order problem (not new exactly, but it looks like you tried to fix related problems in this PR) that I'll write a separate issue for.

File: buildprocess/configureWebpack.js (hunk 1 above)
Comment (kring) [inline buildprocess/configureWebpack.js:5]: Having this line uncommented seems to make my Windows build hang, or at least take a really really long time.

Comment (AlexGilleran) [inline buildprocess/configureWebpack.js:5]: Hmm, _not_ having it makes my build hang :(.

**Pull request merged**